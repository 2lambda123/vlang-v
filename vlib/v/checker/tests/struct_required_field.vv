struct Abc {
	f1 int [required]
	f2 int
	f3 int [required]
}

fn main() {
	_ = Abc{
		f1: 123
		f3: 789
	}
	_ = Abc{
		f1: 123
		f2: 789
	}
	_ = Abc{123, 456}
	_ = Abc{123, 456, 789}
}
