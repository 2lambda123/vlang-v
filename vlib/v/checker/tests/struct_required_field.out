vlib/v/checker/tests/struct_required_field.vv:12:6: error: field `Abc.f3` is required
   10 |         f3: 789
   11 |     }
   12 |     _ = Abc{
      |         ~~~~
   13 |         f1: 123
   14 |         f2: 789
vlib/v/checker/tests/struct_required_field.vv:16:6: error: field `Abc.f3` is required
   14 |         f2: 789
   15 |     }
   16 |     _ = Abc{123, 456}
      |         ~~~~~~~~~~~~~
   17 |     _ = Abc{123, 456, 789}
   18 | }
